python/tempo/tsdf.py

```diff
@@ -157,38 +155,39 @@ def __addPrefixToColumns(self, col_list: list[str], prefix: str) -> "TSDF":
         """
         Add prefix to all specified columns.
         """
-        if prefix != "":
-            prefix = prefix + "_"
+        # no-op if no prefix
+        if not prefix:
+            return self
 
-        df = reduce(
-            lambda df, idx: df.withColumnRenamed(
-                col_list[idx], "".join([prefix, col_list[idx]])
-            ),
-            range(len(col_list)),
-            self.df,
+        # build a column rename map
+        col_map = {col: "_".join([prefix, col]) for col in col_list}
+        # df = self.df.withColumnsRenamed(col_map)
```

Comment:
```suggestion
```

Reply:
I wanted to leave this in as a suggestion for the future. This is a nicer way to implement the rename, but only works in Spark 3.4+

Reply:
I'll clarify in a comment

Reply:
```suggestion
        # In the future (when Spark 3.4+ is standard) we should implement batch rename using:
        # df = self.df.withColumnsRenamed(col_map)
```